src/libraries/System.Net.Quic/tests/FunctionalTests/MsQuicTests.cs

```diff
@@ -788,5 +789,80 @@ public async Task BigWrite_SmallRead_Success(bool closeWithData)
                 }
             }
         }
+
+        [Fact]
+        public async Task BasicTest_WithReadsCompletedCheck()
```

Comment:
The name's like that because it is basically the same test as QuicStreamTests.BasicTest
https://github.com/dotnet/runtime/blob/42af636bde677bef5360c926ae2e7190a2f75830/src/libraries/System.Net.Quic/tests/FunctionalTests/QuicStreamTests.cs#L23-L55
But with added ReadsCompleted checks 😄
In your initial commit, you've added these checks to the BasicTest itself. But that makes it non-fully-basic then 😄 and it had flaky problems with Mock behavior. As we don't prioritize fixing Mock, I moved the test with checks to the test collection that only runs on MsQuic.

That being said, I have no strong feeling in sticking to the name BasicTest. So I can rename it to e.g. `SendReceive_WithReadsCompletedCheck`

Reply:
Ah. I wasn't familiar with BasicTest so this name seemed non-descript. I'll let you decide what to do with it.